common/schema/config.go

```diff
@@ -17,6 +17,27 @@ type Configuration struct {
 	NotMainTableOnly []ColumnKey `validate:"ninterfield=MainTableOnly"`
 	// Materialize lists columns that shall be materialized at ingest instead of computed at query time
 	Materialize []ColumnKey
+	// CustomDictionaries allows enrichment of flows with custom metadata
+	CustomDictionaries map[string]CustomDict `validate:"omitempty,dive"`
+}
+
+// CustomDict represents a single custom dictionary
+type CustomDict struct {
+	Keys       []CustomDictElement `validate:"dive"`
+	Attributes []CustomDictElement `validate:"dive"`
+	Source     string
+	Layout     string `default:"hashed"`
```

Comment:
This is not a recognized metadata keyword. Have a look at `orchestrator/clickhouse/config.go` for `DefaultNetworkSourceConfiguration()` and its use in `init()`.

Reply:
And add `validate:"required,oneof=hashed iptrie complex_key_hashed"` (not 100% sure, but I think this is better to support only some sane choices first).

Reply:
Improved with 542f36cb3cb5c771d1e408eeb8c9360806f7b7a2